java/core/src/main/java/com/microsoft/bond/protocol/StreamNotCloneableException.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond.protocol;
+
+import java.io.IOException;
+
+/**
+ * Indicates that a stream cannot be cloned because the original stream is not cloneable.
+ */
+public class StreamNotCloneableException extends IOException {
```

Comment:
Not going to try to deal with it in this PR, but this is absolutely not an `IOException`. We need to be more deliberate about the exception hierarchy we throw.